### PR TITLE
CI: Replace `_` of in GHA workflow filenames with `-`

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes inconsistency in workflow filenames.

~Docs changes will conflict with #13849, so I'll update them after the PR gets merged.~ WIP

TODO
- [x] Update all occurences in the docs (readmes).
  - CircleCI badge: https://app.circleci.com/settings/project/github/Lightning-AI/lightning/status-badges
  - GHA badge: e.g. https://github.com/Lightning-AI/lightning/actions/workflows/ci-pytorch-test-full.yml

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -80,21 +80,24 @@ ______________________________________________________________________
 
 ## Continuous Integration
 
-Lightning is rigorously tested across multiple GPUs, TPUs CPUs and against major Python and PyTorch versions.
+Lightning is rigorously tested across multiple CPUs, GPUs, TPUs, IPUs, and HPUs and against major Python and PyTorch versions.
```
Comment (akihironitta) [inline README.md:83]: We have nearly the same `/src/pytorch_lightning/README.md` as `/README.md`, but is it on purpose? There was a discrepancy between them, so I was wondering if we could remove/adjust either of them.

**Comment by Borda** [comment]
for me, the `_` and `-` served as grouping so `CI pytorch - test full` become `ci-pytorch_test-full`

**Comment by akihironitta** [comment]
@Borda If you prefer `_` to `-`, we can maybe keep it as is, but I'm favour of using `-` because it's still possible to group them with `-`, and I'm sure people in the team/community will start using them differently between the two, which leads to inconsistency as we currently have `ci-*` and `ci_*`, for example.

**Comment by akihironitta** [comment]
@Borda I'm still in favour of this PR, but feel free to close this PR if you find this PR unreasonable.

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**